sympy/solvers/simplex.py

```diff
@@ -0,0 +1,966 @@
+    d = d[0]
+    _ = lambda x: numbered_symbols(x, start=1)
+    x = Matrix([i for i, j in zip(_('x'), range(n))])
+    yT = Matrix([i for i, j in zip(_('y'), range(m))]).T
+    def ineq(L, r, op):
+        rv = []
+        for r in (op(i, j) for i, j in zip(L, r)):
+            if r == True:
+                continue
+            elif r == False:
+                return [False]
+            if factor:
+                f = factor_terms(r)
+                if f.lhs.is_Mul and f.rhs % f.lhs.args[0] == 0:
+                    assert len(f.lhs.args) == 2, f.lhs
+                    k = f.lhs.args[0]
+                    r = r.func(sign(k)*f.lhs.args[1], f.rhs//abs(k))
+            rv.append(r)
+        return rv
+    eq = lambda x, d: x[0] - d if x else -d
+    F = eq(c*x, d)
+    f = eq(yT*b, d)
+    return (F, ineq(A*x, b, Ge)), (f, ineq(yT*A, c, Le))
+
+
+def _rel_as_nonpos(constr):
+    """return a (np, d, aux) where np is a list of nonpositive
+    expressions that represent the given constraints (possibly rewritten
+    in terms of auxilliary variables) expressible with nonnegative
+    symbols, and d is a dictionary mapping a given symbols to an
+    expression with an auxilliary variable. In some cases a symbol
+    will be used as part of the change of variables, e.g. x: x - z1
+    instead of x: z1 - z2.
+
+    If any constraint is False/empty, return None. Any constraint
+    involving a relationship with oo will create unbounded symbols
+    for all symbols present in the expression, e.g. ``x + y < oo``
+    will allow ``x`` and ``y`` to take on any value. (This is the
+    only type relationship which can be expressed as strict.)
+
+    Examples
+    ========
+
+    >>> from sympy import oo
+    >>> from sympy.solvers.simplex import _rel_as_nonpos
+    >>> from sympy.abc import x, y
+    >>> _rel_as_nonpos([x >= y])
+    ([-x + y], {}, [])
+    >>> _rel_as_nonpos([x > -oo])
+    ([], {x: _z1 - x}, [_z1])
+    >>> _rel_as_nonpos([x >= 3, x <= 5])
+    ([_z1 - 2], {x: _z1 + 3}, [_z1])
+    >>> _rel_as_nonpos([x <= 5])
+    ([], {x: 5 - _z1}, [_z1])
+    >>> _rel_as_nonpos([x >= 1])
+    ([], {x: _z1 + 1}, [_z1])
+    """
+    r = {}  # replacements to handle change of variables
+    np = []  # nonpositive expressions
+    aux = []  # auxilliary symbols added
+    ui = numbered_symbols('z', start=1, cls=Dummy) # auxilliary symbols
+    univariate = {}  # {x: interval} for univariate constraints
+    unbound = set()  # symbols designated as unbound
+
+    # separate out univariates
+    for i in constr:
+        if i == True:
+            continue  # ignore
+        if i == False:
+            return  # no solution
+        i = i.canonical  # +/-oo, if present will be on the rhs
+        if isinstance(i, (Le, Ge, Lt, Gt)) and i.rhs.is_infinite:
+            if isinstance(i, (Lt, Le)) and i.rhs is S.NegativeInfinity:
+                return False
+            if isinstance(i, (Gt, Ge)) and i.rhs is S.Infinity:
+                return False
```

Comment:
Maybe it better to just raise an error if any side is infinite...

Reply:
There is no other way to pass an inequality that allows for any real value. `linprog` uses tuples for limits with None acting as infinity.